File: backend/src/main/java/org/gestionstock/stock/Payload/Response/InventoryItemResponse.java
package org.gestionstock.stock.Payload.Response;


public record InventoryItemResponse(
    String id,
    String name,
    String description,
    Long quantity,
    Double costPrice,
    Double tva,
    ItemCategoryResponse category,
    String createdAt,
    String updatedAt,
    String createdBy,
    String updatedBy
) {
    
}

File: frontend/src/app/models/response/inventory-item-response.ts
import { ItemCategoryResponse } from "./item-category-response";

export interface InventoryItemResponse {
    id: string;
    name: string;
    description: string;
    quantity: number;
    costPrice: number;
    tva: number;
    category: ItemCategoryResponse;
    createdAt: string;
    updatedAt: string;
    createdBy: string;
    updatedBy: string;
}
